lib/builder.js

```diff
@@ -0,0 +1,475 @@
+				}
+
+				// remove notEqual parameters so they aren't processed again
+				filters.mandatory.ne = undefined;
+				filters.mandatory.notEqual = undefined;
+				filters.mandatory.notEqualTo = undefined;
+
+				buildFilters(mustNot).forEach(function (filter) {
+					// right now, only exact match is support
+					if (filter.match) {
+						queryBool['must_not'] = (queryBool['must_not'] || []).concat(filter);
+						return;
+					}
+				});
+			}
+
+
+			buildFilters(filters.mandatory).forEach(function (filter) {
+				if (filter.match) {
+					queryBool.must = (queryBool.must || []).concat(filter);
+
+					return;
+				}
+
+				filterBool.must = (filterBool.must || []).concat(filter);
+			});
+		}
+
+		// add optional filters to the query
+		if (filters && filters.optional) {
+			buildFilters(filters.optional).forEach(function (filter) {
+				if (filter.match) {
+					queryBool.should = (Array.isArray(queryBool.should) ? queryBool.should : [])
+						.concat(filter);
+
+					return;
+				}
+
+				filterBool.should = (Array.isArray(filterBool.should) ? filterBool.should : [])
+					.concat(filter);
+			});
+		}
+
+		// add projection
+		if (filters && filters.field) {
+			// ensure array
+			if (!Array.isArray(filters.field)) {
+				if (/\,/g.test(filters.field)) {
+					filters.field = filters.field.split(/\,/);
+				} else {
+					filters.field = [filters.field];
+				}
+			}
+
+			query._source = (query._source || []).concat(filters.field);
+		}
+
+		// add sorting
+		// NOTE: only works with simple cases as at this stage in the software
+		// the system unable to detect order between asc and desc as originally
+		// passed via the querystring
+		if (queryOptions.sort) {
+
+			if (typeof queryOptions.sort === 'string') {
+				sortOptions = queryOptions.sort.split(/\,/g).map(function (field) {
+					return field.trim();
+				});
+			} else if (Array.isArray(queryOptions.sort) && queryOptions.sort.length) {
+				sortOptions = queryOptions.sort;
+			} else if (typeof queryOptions.sort === 'object') {
+				Object.keys(queryOptions.sort).forEach(function (property) {
+					// support for more traditional way of supplying asc params
+					if (property === 'asc') {
+						sortFields = Array.isArray(queryOptions.sort.asc) ?
+							queryOptions.sort.asc :
+							[queryOptions.sort.asc];
+
+						sortOptions = sortOptions.concat(sortFields);
+						return;
+					}
+
+					// support for more traditional way of supplying desc params
+					if (property === 'desc') {
+						sortFields = Array.isArray(queryOptions.sort.desc) ?
+							queryOptions.sort.desc :
+							[queryOptions.sort.desc];
+
+						sortFields = sortFields.map(function (field) {
+							return '-' + field;
+						});
+
+						sortOptions = sortOptions.concat(sortFields);
+						return;
+					}
+
+					if (queryOptions.sort[property] === 'desc' ||
+						parseInt(queryOptions.sort[property], 10) < 0) {
+						sortOptions.push('-' + property);
+					} else {
```

Comment:
If you put a return in the above `if` statement, you could remove this `else` and have it be the default action on each.